flocker/control/test/test_httpapi.py

```diff
@@ -2769,6 +2769,18 @@ def test_empty(self):
             b"GET", b"/state/datasets", None, OK, response
         )
 
+    def test_unknown_datasets(self):
+        """
+        When the cluster state is ignorant about datasets on a node, the
+        endpoint does not list information for that node.
+        """
+        self.cluster_state_service.apply_changes([
+            NodeState(hostname=u"192.0.2.101", uuid=uuid4(),
+                      manifestations=None)])
```

Comment:
Perhaps set paths to `None` as well since one of these will never be known without the other.